src/Input/Input.js

```diff
@@ -68,6 +73,48 @@ export const styleSheet = createStyleSheet('MuiInput', theme => ({
       // Remove the padding when type=search.
       appearance: 'none',
     },
+    'label + $formControl > &': {
+      '&::-webkit-input-placeholder': {
+        opacity: 0,
+        transition: theme.transitions.create('opacity', {
+          duration: theme.transitions.duration.shorter,
+          easing: theme.transitions.easing.ease,
+        }),
+      },
+      '&::-moz-placeholder': {
+        opacity: 0,
+        transition: theme.transitions.create('opacity', {
+          duration: theme.transitions.duration.shorter,
+          easing: theme.transitions.easing.ease,
+        }),
+      },
+      '&:-ms-input-placeholder': {
+        opacity: 0,
+        transition: theme.transitions.create('opacity', {
+          duration: theme.transitions.duration.shorter,
+          easing: theme.transitions.easing.ease,
+        }),
+      },
+      '&:-moz-placeholder': {
+        opacity: 0,
+        transition: theme.transitions.create('opacity', {
+          duration: theme.transitions.duration.shorter,
+          easing: theme.transitions.easing.ease,
+        }),
+      },
+      '&:focus::-webkit-input-placeholder': {
```

Comment:
@oliviertassinari `JSS` currently doesn't support prefixing `::placeholder` selector. But there is discussion on this issue. Hope the will release it soon.